server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -1861,6 +1870,9 @@ private static Set<ResolvedExpression> expandToOpenClosed(
                 resources.add(new ResolvedExpression(indexAbstraction.getName(), selector));
             } else if (context.isPreserveDataStreams() && indexAbstraction.getType() == Type.DATA_STREAM) {
                 resources.add(new ResolvedExpression(indexAbstraction.getName(), selector));
+            } else if (indexAbstraction.getType() == Type.VIEW) {
+                // a view cannot expand to any indices, return an empty set
+                return Set.of();
```

Comment:
Food for thought: considering that the index name expression resolver is on the critical path, I am wondering if we should pull up this check to the callers and make this an assertion.

By moving this check to: 

https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java#L1721

and 

https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java#L1808

We can avoid some extra steps. I am not sure, if it's worth the optimisation to be honest, but because we had issues in the past, I want to be sure we at least consider it.